### PR TITLE
[11.0][ADD] sale_product_set_variant

### PR DESCRIPTION
**Pull request by simahawk**
Depends on my branch for #574 and forwards #547 (to be updated once this is merged)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please remove the `s` from the module name.

**Comment by simahawk** [comment]
@pedrobaeza done and removed crap from twin PR.

**Comment by pedrobaeza** [comment]
Travis is still red I'm afraid

**Comment by simahawk** [comment]
@pedrobaeza travis green but runbot red. I see a lot of "no such container" [in the log](https://runbot2.odoo-community.org/runbot/static/build/3314636-575-db7381/logs/job_10_test_base.txt).
Any bell?

**Comment by gurneyalex** [comment]
in the build logs: 

Sale Product Set Variants
================
2018-01-12 09:43:06,047 168 WARNING openerp_test odoo.addons.base.module.module: docutils' system message present: <system_message level="2" line="6" source="<string>" type="WARNING"><paragraph>Title underline too short.</paragraph><literal_block xml:space="preserve">Sale Product Set Variants

**Comment by simahawk** [comment]
ouch, thanks, I misread the log... build is finally green! \o/

**Review by TDu — APPROVED** [review]
LGTM
Except for the encoding at the head of the python file, shouldn't they be removed ?

File: sale_product_set_variant/models/product_set_line.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
```
Comment (leemannd) [inline sale_product_set_variant/models/product_set_line.py]: 2017-2018 ?

**Review by leemannd — CHANGES_REQUESTED** [review]
Only nitpicking on copyright dates

File: sale_product_set_variant/models/product_set_line.py (hunk 1 above)
Comment (simahawk) [reply sale_product_set_variant/models/product_set_line.py]: year updated, encoding removed

**Comment by simahawk** [comment]
#574 merged, pending oca dependency removed

File: sale_product_set_variant/wizard/product_set_add.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# Copyright 2018 Camptocamp SA
```
Comment (leemannd) [inline sale_product_set_variant/wizard/product_set_add.py]: 2017-2018

File: sale_product_set_variant/tests/test_product_set.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2018 Camptocamp SA
```
Comment (leemannd) [inline sale_product_set_variant/tests/test_product_set.py]: 2017-2018

File: sale_product_set_variant/models/product_set_line.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2018 Camptocamp SA
```
Comment (leemannd) [inline sale_product_set_variant/models/product_set_line.py]: 2017-2018

**Review by leemannd — APPROVED** [review]
Thanks for the port!
Code review only

File: sale_product_set_variant/wizard/product_set_add.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    def add_set(self):
+        """Add product set, multiplied by quantity in sale order line."""
+        self.ensure_one()
+        so_id = self.env.context.get('active_id')
+        if not so_id:
+            return
+        so = self.env['sale.order'].browse(so_id)
+        max_sequence = 0
+        if so.order_line:
+            max_sequence = max([line.sequence for line in so.order_line])
```
Comment (ghost) [inline sale_product_set_variant/wizard/product_set_add.py:95]: `max(so.order_line.mapped('sequence'))`? :slightly_smiling_face:

**Review by ghost — APPROVED** [review]

**Review by guewen — APPROVED** [review]

**Pull request merged**